coderd/notifications/metrics_test.go

```diff
@@ -493,27 +499,30 @@ func (u *updateSignallingInterceptor) BulkMarkNotificationMessagesFailed(ctx con
 	return u.Store.BulkMarkNotificationMessagesFailed(ctx, arg)
 }
 
-type delayingHandler struct {
+type barrierHandler struct {
 	h notifications.Handler
 
-	delay time.Duration
+	wg *sync.WaitGroup
```

Comment:
suggestion: Could we use a `chan struct{}` here instead?

Reply:
Hm.. we could but can you explain why is it recommended?

Reply:
Fair enough!